crates/sui-graphql-rpc/src/types/address.rs

```diff
@@ -56,8 +61,13 @@ impl Address {
         unimplemented!()
     }
 
-    pub async fn balance(&self, type_: Option<String>) -> Balance {
-        unimplemented!()
+    pub async fn balance(&self, ctx: &Context<'_>, type_: Option<String>) -> Result<Balance> {
+        fetch_balance(
+            ctx.data_unchecked::<sui_sdk::SuiClient>(),
+            &self.address,
```

Comment:
(continued from above) we could pass self.location() here

Reply:
why do a fn call over passing a ref?